packages/core/src/anchor/auth/did-anchor-service-auth.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  AnchorServiceAuth,
+  CeramicApi,
+  DiagnosticsLogger,
+  FetchOpts,
+  fetchJson,
+  FetchRequestParams
+} from '@ceramicnetwork/common'
+import { DagJWS } from 'dids'
+import crypto from 'crypto'
+
+export class DIDAnchorServiceAuth implements AnchorServiceAuth {
+  private _ceramic: CeramicApi
+  private readonly _anchorServiceUrl: string
+  private readonly _logger: DiagnosticsLogger
+
+  constructor(
+    anchorServiceUrl: string,
+    logger: DiagnosticsLogger,
+  ) {
+    this._anchorServiceUrl = anchorServiceUrl
+    this._logger = logger
+  }
+
+  /**
+   * Set Ceramic API instance
+   *
+   * @param ceramic - Ceramic API used for various purposes
+   */
+  set ceramic(ceramic: CeramicApi) {
+    this._ceramic = ceramic
+  }
+
+  init = async (): Promise<void> => {
+    return
+  }
+
+  sendAuthenticatedRequest = async (url: URL | string, opts?: FetchOpts): Promise<any> => {
+    if (!this._ceramic) {
+      throw new Error('Missing Ceramic instance required by this auth method')
+    }
+    const { request } = await this.signRequest({url, opts})
+    return await this._sendRequest(request)
+  }
+
+  async signRequest(request: FetchRequestParams): Promise<{request: FetchRequestParams, jws: DagJWS}> {
+    const payload: any = { url: request.url, nonce: crypto.randomUUID() }
+    if (request.opts) {
+      if (request.opts.body) {
+        payload.body = request.opts.body
+      }
```

Comment:
```suggestion
      if (request.opts.body) {
        const car = CarFactory.fromBytes(u8a.fromString(request.opts.body, 'binary'))
        payload.body = car.roots[0].toString()
      }
```
There is only a body if a CAR file is present (assuming this PR is on top of CAR file PR). This way we sign the CID of the root.

Using https://github.com/ceramicnetwork/cartonne

Reply:
ty! I actually wonder if using a hash over the body in general is a better approach than including the full body as you mentioned

Reply:
It could be, but in the case of a CAR file that would be redundant since the root hash is already a hash over the entire body.